portal/sign/assets/signit.php

```diff
@@ -0,0 +1,47 @@
+<?php
```

Comment:
Don't forget the author/license header :)

Reply:
Where is this script being called? (trying to search for it in the PR and can't find where it is called)

Reply:
It runs as a separate app when you add to the apps list options.
![image](https://user-images.githubusercontent.com/1859985/62919296-41d55e80-bd70-11e9-9d8c-91af901e395e.png)
This way it can be run on a separate tablet (remote). Communication between forms/templates and the signature pad is via message channels.

Reply:
Lets add this to the apps list then (turned off of course :) )

Reply:
I've got know idea what a message channel is, and just looked it up:
https://developer.mozilla.org/en-US/docs/Web/API/Channel_Messaging_API/Using_channel_messaging

Reply:
So to clarify.
1. This script you login via the app with the openemr user.
2. The script basically kills the session after the successful login. Would the following make more sense there:
```
require_once(dirname(__FILE__) . "/../../src/Common/Session/SessionUtil.php");
OpenEMR\Common\Session\SessionUtil::coreSessionDestroy();
```
3. Then it just hangs out waiting for message to come in?

Reply:
I looked through the signer_api.js for where remoteDevice is used, which made things more clear in my brain. Very neat stuff.

Reply:
Yep, it's automatic ie if openemr form signature tag is clicked, the app sends a request to the remote and waits for signature the passes signature back to form.
For sure I will use the new session manager stuff and this thread explains how remote works
https://community.open-emr.org/t/adding-patient-portal-document-templates-to-openemr-patient-documents/12141

Reply:
In regards to session/cookie, this should work well when getting to this script through the standard login app for it (ie. will get is own session). Any thoughts about what would happen if:
1. A user does a refresh on this screen?
2. A user directly goes to this screen via a hard link copy/paste when logged into core OpenEMR?

Reply:
(note above doesn't need to delay getting this into codebase and is something can refine once it's in the codebase)

Reply:
1. We still have main.php as top and csrf token so refresh behaves as normal openemr i.e asks to leave.
2. Will fail to init signer api. Turns out when destroying the remote signit session after login will break other logins. Also, because main starts and timeout timer runs after signit login, destroying remotes session causes unpredictable results.

I still believe calling restore session from these timer routines breaks sessions